src/main/java/org/openlmis/notification/web/usercontactdetails/UserContactDetailsController.java

```diff
@@ -169,6 +176,53 @@ public UserContactDetailsDto saveUpdateUserContactDetails(
     return toDto(contactDetails);
   }
 
+  /**
+   * Creates or updates contact details for multiple users.
+   *
+   * @param userContactDetailsDtoList  list of user contact details
+   * @return  the new or updated contact details for list of users
+   */
+  @PutMapping("/userContactDetails/batch")
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public UserContactDetailsResponseDto saveUsersContactDetails(
+      @RequestBody List<UserContactDetailsDto> userContactDetailsDtoList) {
+    List<UserContactDetailsResponseDto.UserDetailsResponse> successfulResults = new ArrayList<>();
```

Comment:
I'm wondering if this whole body, maybe with an exception of a role check, should be moved to a Service.